src/planner/binder/tableref/bind_basetableref.cpp

```diff
@@ -184,11 +184,7 @@ unique_ptr<BoundTableRef> Binder::Bind(BaseTableRef &ref) {
 		}
 	}
 	if (!table_or_view) {
-		string table_name = ref.catalog_name;
-		if (!ref.schema_name.empty()) {
-			table_name += (!table_name.empty() ? "." : "") + ref.schema_name;
-		}
-		table_name += (!table_name.empty() ? "." : "") + ref.table_name;
+		string table_name = ParseInfo::QualifierToStringNoQuotes(ref.catalog_name, ref.schema_name, ref.table_name);
```

Comment:
This is a work-around for replacement scans not having a schema/catalog - can we instead extend the replacement scans with schema/catalog and remove this code entirely?